engine/src/main/java/io/zeebe/engine/nwe/event/BoundaryEventProcessor.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.engine.nwe.event;
+
+import io.zeebe.engine.nwe.BpmnElementContext;
+import io.zeebe.engine.nwe.BpmnElementProcessor;
+import io.zeebe.engine.nwe.BpmnProcessingException;
+import io.zeebe.engine.nwe.behavior.BpmnBehaviors;
+import io.zeebe.engine.nwe.behavior.BpmnIncidentBehavior;
+import io.zeebe.engine.nwe.behavior.BpmnStateBehavior;
+import io.zeebe.engine.nwe.behavior.BpmnStateTransitionBehavior;
+import io.zeebe.engine.nwe.behavior.BpmnVariableMappingBehavior;
+import io.zeebe.engine.processor.workflow.deployment.model.element.ExecutableBoundaryEvent;
+
+public final class BoundaryEventProcessor implements BpmnElementProcessor<ExecutableBoundaryEvent> {
+
+  private final BpmnStateBehavior stateBehavior;
+  private final BpmnStateTransitionBehavior stateTransitionBehavior;
+  private final BpmnVariableMappingBehavior variableMappingBehavior;
+  private final BpmnIncidentBehavior incidentBehavior;
+
+  public BoundaryEventProcessor(final BpmnBehaviors bpmnBehaviors) {
+    stateBehavior = bpmnBehaviors.stateBehavior();
+    stateTransitionBehavior = bpmnBehaviors.stateTransitionBehavior();
+    variableMappingBehavior = bpmnBehaviors.variableMappingBehavior();
+    incidentBehavior = bpmnBehaviors.incidentBehavior();
+  }
+
+  @Override
+  public Class<ExecutableBoundaryEvent> getType() {
+    return ExecutableBoundaryEvent.class;
+  }
+
+  @Override
+  public void onActivating(
+      final ExecutableBoundaryEvent element, final BpmnElementContext context) {
+    // the boundary event is already triggered when the activating event is written
+
+    stateTransitionBehavior.transitionToActivated(context);
+  }
+
+  @Override
+  public void onActivated(final ExecutableBoundaryEvent element, final BpmnElementContext context) {
+
+    stateTransitionBehavior.transitionToCompleting(context);
+  }
+
+  @Override
+  public void onCompleting(
+      final ExecutableBoundaryEvent element, final BpmnElementContext context) {
+
+    variableMappingBehavior
+        .applyOutputMappings(context, element)
+        .ifRightOrLeft(
+            ok -> stateTransitionBehavior.transitionToCompleted(context),
+            failure -> incidentBehavior.createIncident(failure, context));
+  }
+
+  @Override
+  public void onCompleted(final ExecutableBoundaryEvent element, final BpmnElementContext context) {
+
+    stateTransitionBehavior.takeOutgoingSequenceFlows(element, context);
```

Comment:
Is the flownode that the boundary event is attached to already terminated when the boundary event completed? I would expect some logic here to check whether the boundary event is interrupting or not.

Reply:
This logic is in the activity itself. When the `activating` event is written for the boundary event then it was already triggered and the activity was terminated - if it is an interrupting boundary event. So, there is no much to do here.